controllers/activegate/reconciler/capability/service.go

```diff
@@ -12,23 +12,40 @@ import (
 	"k8s.io/apimachinery/pkg/util/intstr"
 )
 
+const AnnotationEnableStatsD = dynatracev1beta1.InternalFlagPrefix + "enable-statsd"
+
 func createService(instance *dynatracev1beta1.DynaKube, feature string) *corev1.Service {
+	enableStatsD := instance.FeatureEnableStatsDIngest()
+	ports := []corev1.ServicePort{
+		{
+			Name:       consts.ServicePortName,
+			Protocol:   corev1.ProtocolTCP,
+			Port:       consts.ServicePort,
+			TargetPort: intstr.FromString(consts.ServiceTargetPort),
+		},
+	}
+	if enableStatsD {
+		ports = append(ports, corev1.ServicePort{
+			Name:       consts.StatsDIngestPortName,
+			Protocol:   corev1.ProtocolUDP,
+			Port:       consts.StatsDIngestPort,
+			TargetPort: intstr.FromString(consts.StatsDIngestTargetPort),
+		})
+	}
+
 	return &corev1.Service{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      BuildServiceName(instance.Name, feature),
 			Namespace: instance.Namespace,
 			Labels:    statefulset.BuildLabelsFromInstance(instance, feature),
+			Annotations: map[string]string{
```

Comment:
Is this feature flag used on the service? It should only be set on resources that are controlled by the user (Dynakube, ...)

Reply:
That's not a feature flag, only an internal annotation (`internal.operator.dynatrace.com/...`) used to control service features before even the dynakube is updated. As such, it was never meant to be edited by hand.

It was a long time ago but if I remember correctly I introduced it to make service comparisons easier and faster. As an alternative, I can imagine comparing service Specs with `reflect.DeepEqual()`, does that sound better?